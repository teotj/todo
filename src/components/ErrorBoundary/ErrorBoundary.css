$color1: #900;
$color2: #800;

.root {
  padding: 5px;
  background: repeating-linear-gradient(135deg, $color1, $color1 10px, $color2 10px, $color2 20px);
  color: #fff;
  text-align: left;
  line-height: 14px;
}

.name {
  font-size: 12px;
  font-weight: 500;
  margin-bottom: 5px;
}

.message {
  font-size: 10px;
  font-weight: normal;
}
