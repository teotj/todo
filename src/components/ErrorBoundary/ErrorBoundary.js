import React, { Component } from 'react';
import PropTypes from 'prop-types';

import css from './ErrorBoundary.css';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false, error: null };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true, error };
  }

  render() {
    const { children } = this.props;
    const { hasError, error } = this.state;

    if (hasError) {
      return (
        <div className={css.root}>
          <div className={css.name}>{error.name}</div>
          <div className={css.message}>{error.message}</div>
        </div>
      );
    }

    return children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired
};

export default ErrorBoundary;
