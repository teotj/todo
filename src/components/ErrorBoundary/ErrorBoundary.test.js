import React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundary from './ErrorBoundary';

const renderErrorBoundary = (props = {}) => <ErrorBoundary {...props} />;

const shallowRenderErrorBoundary = (props) => shallow(renderErrorBoundary(props));

describe('<ErrorBoundary>', () => {
  it('Should render', () => {
    const actual = shallowRenderErrorBoundary({ children: 'OK' });
    expect(actual).toMatchSnapshot();
  });

  it('Should render an error', () => {
    const actual = shallowRenderErrorBoundary({ children: 'OK' });
    const error = {
      name: 'Error name',
      message: 'Error message'
    };
    actual.simulateError(error);
    expect(actual).toMatchSnapshot();
  });
});
