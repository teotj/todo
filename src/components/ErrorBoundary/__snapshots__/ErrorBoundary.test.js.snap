// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> Should render 1`] = `"OK"`;

exports[`<ErrorBoundary> Should render an error 1`] = `
<div
  className="root"
>
  <div
    className="name"
  >
    Error name
  </div>
  <div
    className="message"
  >
    Error message
  </div>
</div>
`;
